### ISSUE TITLE
Audit based Hardening Policy not giving Alerts upon the Violations - "System and Information Integrity - System Monitoring make files under /dev/"

### ISSUE THREAD

**Issue by PriyaAccuknox** [kubearmor/KubeArmor#1397]
Bug 1:  The Given Audit based Hardening Policy not giving Alerts upon the Violations - "System and Information Integrity - System Monitoring make files under /dev/"

```sh
apiVersion: security.kubearmor.com/v1
kind: KubeArmorPolicy
metadata:
  name: harden-wordpress-write-under-dev-dir
  namespace: wordpress-mysql
spec:
  action: Audit
  file:
    matchDirectories:
    - dir: /dev/
      readOnly: true
      recursive: true
  message: Alert! File creation under /dev/ directory detected.
  selector:
    matchLabels:
      app: wordpress
  severity: 5
  tags:
  - NIST
  - NIST_800-53_AU-2
  - NIST_800-53_SI-4
  - MITRE
  - MITRE_T1036_masquerading
```

**General Information**

- Environment description  -GKE
- Linux cs-610260858826-default 5.15.120+ #1 SMP Wed Aug 9 09:14:11 UTC 2023 x86_64 GNU/Linux
- Orchestration system version - Client Version: v1.27.4
                                                     Kustomize Version: v5.0.1
                                                     Server Version: v1.27.3-gke.100

- Target pods - wordpress, nginx
- Deployment  Details
```sh 
shanmuga@cloudshell:~ (centering-study-396808)$ kubectl describe deployment wordpress -n wordpress-mysql 
Name:                   wordpress
Namespace:              wordpress-mysql
CreationTimestamp:      Wed, 30 Aug 2023 09:38:49 +0000
Labels:                 app=wordpress
Annotations:            deployment.kubernetes.io/revision: 1
Selector:               app=wordpress
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:  app=wordpress
  Containers:
   wordpress:
    Image:      wordpress:4.8-apache
    Port:       80/TCP
    Host Port:  0/TCP
    Environment:
      WORDPRESS_DB_HOST:      mysql
      WORDPRESS_DB_PASSWORD:  root-password
    Mounts:                   <none>
  Volumes:                    <none>
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Progressing    True    NewReplicaSetAvailable
  Available      True    MinimumReplicasAvailable
OldReplicaSets:  <none>
NewReplicaSet:   wordpress-7c966b5d85 (1/1 replicas created)
Events:          <none>
```

```sh
shanmuga@cloudshell:~ (centering-study-396808)$ kubectl exec -it wordpress-7c966b5d85-lm5fl -n wordpress-mysql -- bash
root@wordpress-7c966b5d85-lm5fl:/var/www/html# cd /dev
root@wordpress-7c966b5d85-lm5fl:/dev# touch priya
root@wordpress-7c966b5d85-lm5fl:/dev# ls
core  fd  full  mqueue  null  priya  ptmx  pts  random  shm  stderr  stdin  stdout  termination-log  tty  urandom  zero
root@wordpress-7c966b5d85-lm5fl:/dev# touch priya123
root@wordpress-7c966b5d85-lm5fl:/dev# cd core
bash: cd: core: Not a directory
root@wordpress-7c966b5d85-lm5fl:/dev# mkdir shan
root@wordpress-7c966b5d85-lm5fl:/dev# cat shm
cat: shm: Is a directory
root@wordpress-7c966b5d85-lm5fl:/dev# cd shm
root@wordpress-7c966b5d85-lm5fl:/dev/shm# ls
root@wordpress-7c966b5d85-lm5fl:/dev/shm# touch kngofnd
```

 
```sh
shanmuga@cloudshell:~ (centering-study-396808)$ karmor logs
local port to be used for port forwarding kubearmor-relay-5ccb6b6ffb-5rlq5: 32897 
Created a gRPC client (localhost:32897)
Checked the liveness of the gRPC server
Started to watch alerts
^Z
[2]+  Stopped                 karmor logs
```

**Comment by PriyaAccuknox** [comment]
@nyrahul

**Comment by daemon1024** [comment]
Fixed in #1542

**Comment by daemon1024** [comment]
Some issue with policy matching, it's only raising alerts when you are doing a read operation
![image](https://github.com/kubearmor/KubeArmor/assets/47106543/417ec9f6-38de-4a44-897d-74f673c0d3e6)

We will check what's the issue.
